### PR TITLE
[Backend] Introduce Compiler class

### PR DESCRIPTION
**Pull request by YongseopKim**
Let's introduce Compiler class.

ONE-DCO-1.0-Signed-off-by: Yongseop Kim <<yons.kim@samsung.com>>

### CODE REVIEW

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by hyunsik-yoon — APPROVED** [review]

**Comment by jyoungyun** [comment]
I will wait for #490, #488 PR to be merged.
After merged these PRs, please uncomment your codes.

**Review by jyoungyun — APPROVED** [review]
LGTM

@YongseopKim intends to uncomment and use it whenever the user needs. It will be reflected as it is.

**Pull request merged**
